File: LeetCode/first-unique-char.cpp
int firstUniqChar(string s) {
	int arr[26] = { 0 };
	for (auto& a : s) {
		arr[a - 'a']++;
	}
	for (int i = 0; i < s.length(); i++) {
		if (arr[s[i] - 'a'] == 1) return i;
	}
	return -1;
}

File: LeetCode/first_unique_char.java
public int firstUniqChar(String s) {
    if (s == null || s.length() == 0) return -1;
    HashMap<Character, Integer> map = new HashMap<>();
    for (int i = 0; i < s.length(); i++) {
        if (!map.containsKey(s.charAt(i))) {
            map.put(s.charAt(i), 1);
        } else map.put(s.charAt(i), map.get(s.charAt(i)) + 1);
    }
    for (int i = 0; i < s.length(); i++) {
        char c = s.charAt(i);
        if (map.get(c) == 1) {
            return i;
        }
    }
    return -1;
}
